src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -247,11 +253,36 @@ private static ChatMessage FromOllamaMessage(OllamaChatResponseMessage message)
         return new ChatMessage(new(message.Role), contents);
     }
 
+    private static FunctionCallContent ToFunctionCallContent(OllamaFunctionToolCall function)
+    {
+        var id = Guid.NewGuid().ToString().Substring(0, 8);
```

Comment:
Nit: on core this could be:
```C#
var id = RandomNumberGenerator.GetHexString(8); // with `, lowercase = true` if we care about that
```

Reply:
Is there anything preventing us from using a full Guid for the identifier?

Reply:
> Is there anything preventing us from using a full Guid for the identifier?

We're copying the convention used in OpenAI's function calling convention here, since Ollama doesn't have a clear convention of its own.

My intuition (which I haven't actually confirmed) is that it's the LLM that's matching the function call results to entries in the set of prior function calls, and hence shorter values are more likely to work with it being innately fuzzy. We benefit from the ID being unique, but disbenefit from it being long. 8 chars is the tradeoff that OpenAI chose and we're reusing that decision here. Again, I haven't actually tested this intuition in any systematic way, but doing something we know works is reasonably safe.

Reply:
To be honest I'm not even sure that Ollama uses the call IDs at all, but it's not harmful to make them available like this.